internal/appland/client.go

```diff
@@ -12,11 +12,31 @@ import (
 	"github.com/applandinc/appland-cli/internal/config"
 )
 
-type Client struct {
+type Client interface {
+	BuildUrl(paths ...interface{}) string
+	Context() *config.Context
+	CreateMapSet(app, org string, scenarios []string) (*CreateMapSetResponse, error)
+	CreateScenario(org string, scenarioData io.Reader) (*ScenarioResponse, error)
+	DeleteAPIKey() error
+	Login(login string, password string) error
+	TestAPIKey(apiKey string) (bool, error)
+	delete(url string, body io.Reader) (*http.Response, error)
+	get(url string, body io.Reader) (*http.Response, error)
+	newAuthRequest(method, url string, body io.Reader) (*http.Request, error)
+	post(url string, body io.Reader) (*http.Response, error)
+}
```

Comment:
What is the benefit or cause of defining this as an interface? I wouldn't expect that we need many implementations.

Reply:
Defining it as an interface allows it to be mocked easily. We need two implementations: the real one, and one for testing. See `cmd/login_test.go`.